### PR TITLE
applications: nrf5340_audio: Bad_frame flag for wrong sizes

### PR DESCRIPTION
**Pull request by alexsven**
- Set bad_frame flag if frame_size is not matching desired_size

### CODE REVIEW

**Review by koffes — APPROVED** [review]

File: applications/nrf5340_audio/src/audio/le_audio_rx.c [hunk 1]
```diff
@@ -57,7 +57,11 @@ void le_audio_rx_data_handler(uint8_t const *const p_data, size_t data_size, boo
 						    AUDIO_SYNC_TIMER_CURR_TIME_CAPTURE_CHANNEL);
 
 	rx_stats[channel_index].recv_cnt++;
-	if (data_size != desired_data_size && !bad_frame) {
+	if (data_size != desired_data_size) {
+		/* A valid frame should always be equal to desired_data_size, set bad_frame
+		 * if that is not the case
+		 */
+		bad_frame = true;
 		data_size_mismatch = true;
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/audio/le_audio_rx.c:65]: data_size_mismatch is not used now?

**Review by erikrobstad — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/audio/le_audio_rx.c (hunk 1 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/audio/le_audio_rx.c:65]: Right you are, removed

**Review by erikrobstad — CHANGES_REQUESTED** [review]
data_size_mismatch is still there.
You probably forgot to commit?

**Comment by alexsven** [comment]
> data_size_mismatch is still there. You probably forgot to commit?

Correct, again! 'Removed' 2.0 ;)

**Review by erikrobstad — APPROVED** [review]

**Pull request merged**